packages/connector-tencent-sms/src/index.ts

```diff
@@ -0,0 +1,115 @@
+import type {
+  CreateConnector,
+  GetConnectorConfig,
+  SendMessageFunction,
+  SmsConnector,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorType,
+  validateConfig,
+} from '@logto/connector-kit';
+import { assert } from '@silverhand/essentials';
+import { HTTPError } from 'got';
+
+import { isSmsErrorResponse, sendSMSRequest } from '@/http';
+import type { TencentSmsConfig } from '@/schema';
+import { SmsConfigGuard, tencentErrorResponse } from '@/schema';
+
+import { defaultMetadata } from './constant';
+
+function safeGetArray<T>(value: Array<T | undefined>, index: number): T {
+  const item = value[index];
+
+  assert(
+    item,
+    new ConnectorError(ConnectorErrorCodes.General, `Cannot find item at index ${index}`)
+  );
+
+  return item;
+}
+
+function sendMessage(getConfig: GetConnectorConfig): SendMessageFunction {
+  return async (data, inputConfig) => {
+    const { to, type, payload } = data;
+    const config = inputConfig ?? (await getConfig(defaultMetadata.id));
+    validateConfig<TencentSmsConfig>(config, SmsConfigGuard);
+    const { accessKeyId, accessKeySecret, signName, templates, sdkAppId, region } = config;
+    const template = templates.find(({ usageType }) => usageType === type);
+
+    assert(
+      template,
+      new ConnectorError(
+        ConnectorErrorCodes.TemplateNotFound,
+        `Cannot find template for type: ${type}`
+      )
+    );
+
+    try {
+      const httpResponse = await sendSMSRequest(template.templateCode, [payload.code], to, {
+        secretId: accessKeyId,
+        secretKey: accessKeySecret,
+        sdkAppId,
+        region,
+        signName,
+      });
+
+      const { body: responseData } = httpResponse;
+      const isError = isSmsErrorResponse(responseData);
+
+      if (isError) {
+        const { Response } = responseData;
+        const { Error } = Response;
+        throw new ConnectorError(ConnectorErrorCodes.General, `${Error.Code}: ${Error.Message}`);
+      }
+
+      const {
+        Response: { SendStatusSet, RequestId },
+      } = responseData;
+
+      const { Code, Message } = safeGetArray(SendStatusSet, 0);
+
+      assert(
+        Code.toLowerCase() === 'ok',
+        new ConnectorError(
+          ConnectorErrorCodes.General,
+          `${Code}: ${Message}, RequestId: ${RequestId}`
+        )
+      );
+
+      return httpResponse;
+    } catch (error: unknown) {
+      if (!(error instanceof HTTPError)) {
+        throw error;
+      }
+
+      const {
+        response: { body: rawBody },
+      } = error;
+
+      assert(typeof rawBody !== 'object', new ConnectorError(ConnectorErrorCodes.InvalidResponse));
```

Comment:
This can be saved, as we are using zod guard to parse the `rawBody`

Reply:
adjusted